proj/src/main/webapp/script.js

```diff
@@ -46,9 +99,13 @@ if(btnSignUp){
                 firebase.database().ref("users/" + auth.currentUser.uid).set({
                     firstName : fname.value,
                     lastName : lname.value,
-                    tags : tagStr.value.split(',')
+                    tags : tagList
```

Comment:
In addition to storing what users are in each chat, you can store chat keys or something in the user. Or a map from tags to the chat keys so that you don't need to go through all the users for each chat. This might be what arrington already said.

Reply:
ya i'll get there soon